ubuntu_flutter_template/hooks/pre_gen.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:io';
+import 'package:mason/mason.dart';
+import 'package:path/path.dart' as path;
+import 'package:pub_api_client/pub_api_client.dart';
+
+import 'utils.dart';
+
+Future<void> run(HookContext context) async {
+  final directory = Directory.current;
+  final directoryName = path.basename(directory.path);
+  final userDefinedWorkspaceName = context.vars['workspaceName'] as String;
+  final name = userDefinedWorkspaceName.isNotEmpty
+      ? userDefinedWorkspaceName
+      : directoryName;
+  context.vars['workspaceName'] = name;
+
+  if (directory.listSync().isNotEmpty) {
+    // TODO: Print pwd
+    context.logger.warn('The current directory is not empty.');
+    final shouldContinue = context.logger.confirm('Do you want to continue?');
+    if (!shouldContinue) {
+      exit(1);
+    }
+  }
+
+  maybeInstallMelos(context);
+  maybeInstallFvm(context);
+
```

Comment:
We probably need to also initialize fvm in the workspace directory (i.e. create an .fvmrc with the corresponding flutter version). Currently I'm getting
```
melos.yaml: SDK path is not valid. Could not find dart tool at /tmp/testproject/.fvm/flutter_sdk/bin/dart
```
when running
```
mason make ubuntu_flutter_template
```

Reply:
Hmm, that's odd, we are initializing fvm here: https://github.com/ubuntu/flutter_project_template/blob/main/ubuntu_flutter_template/hooks/post_gen.dart#L21
I'm guessing it crashes silently for you then?

Reply:
So you not get any fvm output before the crash?

Reply:
Aaaaah, the .fvmrc file isn't pushed since git ignore dot files automatically when adding a directory